torchaudio/functional.py

```diff
@@ -1287,15 +1287,26 @@ def phaser(
     delay_pos = 0
     mod_pos = 0
 
-    output_waveform = torch.zeros_like(waveform, dtype=dtype, device=device)
+    # TODO: Autograd support?
+    output_waveform_pre_gain_list = []
+    waveform = waveform * gain_in
+    delay_buf = delay_buf * decay
+    waveform_list = [waveform[:, i] for i in range(waveform.size(1))]
+    delay_buf_list = [delay_buf[:, i] for i in range(delay_buf.size(1))]
+    mod_buf_list = [mod_buf[i] for i in range(mod_buf.size(0))]
 
     for i in range(waveform.shape[-1]):
-        idx = int((delay_pos + mod_buf[mod_pos]) % delay_buf_len)
-        temp = (waveform[:, i] * gain_in) + (delay_buf[:, idx] * decay)
+        idx = int((delay_pos + mod_buf_list[mod_pos]) % delay_buf_len)
         mod_pos = (mod_pos + 1) % mod_buf_len
         delay_pos = (delay_pos + 1) % delay_buf_len
-        delay_buf[:, delay_pos] = temp
-        output_waveform[:, i] = temp * gain_out
+        temp = (waveform_list[i]) + (delay_buf_list[idx])
```

Comment:
This could be parallelized delay_buf_len entries at a time

Reply:
The dependency on the previous result is what, of course, makes this not applicable to be executed by one of our convolution algorithms. However, maybe there would be value in some kind of [filter](https://en.wikipedia.org/wiki/Filter_(signal_processing)) abstraction that does this generic and efficiently? We are certainly not foreign to these due to [RNNs](https://en.wikipedia.org/wiki/Recurrent_neural_network).